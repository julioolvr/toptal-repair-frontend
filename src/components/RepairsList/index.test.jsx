import React from 'react';
import { shallow } from 'enzyme';

import { RepairsList } from '.';

it('renders an empty list of repairs', () => {
  const wrapper = shallow(<RepairsList user={{ manager: true }} />);
  expect(wrapper).toMatchSnapshot();
});

it("allows the user to add a repair if they're a manager", () => {
  const wrapper = shallow(<RepairsList user={{ manager: true }} />);
  expect(wrapper.find('EmptyRepairsList').prop('canAddRepairs')).toBe(true);
});

it("prevents the user from adding repair if they're not a manager", () => {
  const wrapper = shallow(<RepairsList user={{ manager: false }} />);
  expect(wrapper.find('EmptyRepairsList').prop('canAddRepairs')).toBe(false);
});

it('renders a list of repairs with content', () => {
  const repairs = [{ id: 1, title: 'Test repair 1' }, { id: 1, title: 'Test repair 1' }];
  const wrapper = shallow(<RepairsList user={{ manager: true }} repairs={repairs} />);
  expect(wrapper).toMatchSnapshot();
});
