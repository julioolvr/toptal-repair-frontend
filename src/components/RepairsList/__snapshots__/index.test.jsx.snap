// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of repairs with content 1`] = `
<div
  className="RepairsList"
>
  <RepairRow
    repair={
      Object {
        "id": 1,
        "title": "Test repair 1",
      }
    }
  />
  <RepairRow
    repair={
      Object {
        "id": 1,
        "title": "Test repair 1",
      }
    }
  />
</div>
`;

exports[`renders an empty list of repairs 1`] = `
<div
  className="RepairsList"
>
  <EmptyRepairsList />
</div>
`;
